python/reg_interface/common/sca_utils.py

```diff
@@ -41,6 +41,13 @@ def fpga_keep_hard_reset(ohList):
     subheading('Asserting FPGA Hard Reset (and keeping it in reset)')
     sendScaCommand(ohList, 0x2, 0x10, 0x4, 0x0, False)
 
+def fpga_remove_hard_reset(ohList):
+    subheading('Disabling monitoring')
+    writeReg(getNode('GEM_AMC.SLOW_CONTROL.SCA.ADC_MONITORING.MONITORING_OFF'), 0xffffffff)
```

Comment:
This alters the monitoring, why?  Moreover after issuing the `sendSCACommand` at line 49 why doesn't it restore monitoring back to the original value?

Ideally it should:

1. read the setting of `GEM_AMC.SLOW_CONTROL.SCA.ADC_MONITORING.MONITORING_OFF` and store it in a temp value,
2. write `0xffffffff` to that register,
3. perform the `sendScaCommand` as shown, and then
4. restore `GEM_AMC.SLOW_CONTROL.SCA.ADC_MONITORING.MONITORING_OFF` to the original stored value.

Reply:
I agree, but this was always the case, so it's not related to this pull request.
We never used monitoring, so I just disabled it before running most of the SCA functions. But I agree that it should be read first, then disabled, and then restored to the original value at the end of a given function.

Reply:
Please implement steps 1 through 4 for this function `fpga_remove_hard_reset`.

Reply:
Fine, I just pushed it. But it's completely untested, please test it.